subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Output directories inside build dir
+set(src_out ${ZEPHYR_BINARY_DIR}/source/generated)
+set(include_out ${ZEPHYR_BINARY_DIR}/include/generated)
+
+# Make sure that output directory for *.c files exist
+file(MAKE_DIRECTORY ${src_out})
+
+# This file is used as the source for the parser generator
+set(cddl_file ${CMAKE_CURRENT_LIST_DIR}/modem_update.cddl)
+
+# These are the entry types needed by the source code
+set(entry_types Wrapper Sig_structure1 Segments)
+
+set(decode_c_name modem_update_decode.c)
+set(decode_h_name modem_update_decode.h)
+set(types_h_name modem_update_types.h)
+set(decode_c_tmp ${src_out}/${decode_c_name}.tmp)
+set(decode_h_tmp ${include_out}/${decode_h_name}.tmp)
+set(types_h_tmp ${include_out}/${types_h_name}.tmp)
+set(decode_c ${src_out}/${decode_c_name})
+set(decode_h ${include_out}/${decode_h_name})
+set(types_h ${include_out}/${types_h_name})
+set(src_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
+set(license ${CMAKE_CURRENT_LIST_DIR}/license.txt)
+
+function(cat IN_FILE OUT_FILE)
+  file(READ ${IN_FILE} CONTENTS)
+  file(APPEND ${OUT_FILE} "${CONTENTS}")
+endfunction()
+
+foreach(target decode_c decode_h types_h)
+  list(APPEND CAT_COMMANDS COMMAND cat ${license} ${${target}_tmp} >> ${target})
+  list(APPEND CF_COMMANDS COMMAND clang-format -i ${${target}})
+endforeach()
+
+add_custom_command(
+  OUTPUT ${decode_c} ${decode_h} ${types_h}
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_CDDL_GEN_MODULE_DIR}/scripts/cddl_gen.py
+  -i ${cddl_file}
+  --oc ${decode_c_tmp}
+  --oh ${decode_h_tmp}
+  --oht ${types_h_tmp}
```

Comment:
```suggestion
  --oc ${decode_c}
  --oh ${decode_h}
  --oht ${types_h}
```
Passing the temp files here messes with the `#include` statements in the generated files, and I'm pretty sure you don't need them